cli/commands/project_config.go

```diff
@@ -80,22 +81,55 @@ func NewProjectConfigFromFile(filename string) *ProjectConfig {
 
 // Ensures our configuration data structure conforms to our ad hoc schema.
 // @TODO do this in a more formal way. See docker/libcompose for an example.
-func ValidateProjectConfig(config *ProjectConfig) error {
-	if len(config.Version) == 0 {
+func (c *ProjectConfig) ValidateConfigVersion() error {
+	if len(c.Version) == 0 {
 		return fmt.Errorf("No 'version' property detected.")
 	}
 
-	if config.Version != "1.0" {
-		return fmt.Errorf("Version '1.0' is the only supported value, found '%s'.", config.Version)
+	if c.Version != "1.0" {
+		return fmt.Errorf("Version '1.0' is the only supported value, found '%s'.", c.Version)
 	}
 
-	if config.Scripts != nil {
-		for id, script := range config.Scripts {
-			if len(script.Run) > 10 {
-				util.Logger().Warning.Printf("Project script '%s' has more than 10 run items (%d). You should create a shell script to contain those.", id, len(script.Run))
+	return nil
+}
+
+// Validate the config scripts against a set of rules/norms
+func (c *ProjectConfig) ValidateProjectScripts(subcommands []cli.Command) {
```

Comment:
That's a nicetohave, but it does make sense to make sure we don't cross the script/alias streams as an enforced clarity practice.